### ISSUE TITLE
Very large number gets reformated as invalid JSON

### ISSUE THREAD

**Issue by j-brooke** [j-brooke/FracturedJson#27]
As revealed in [this issue](https://github.com/j-brooke/FracturedJsonJs/issues/10), when attempting to justify a number list, FracturedJson will sometimes write very large numbers as "Infinity", without quotes, resulting in invalid JSON output.

The numbers for which this happens are ones that overflow a 64-bit float's range - for example, `1e500`.  This happens when FJ tries to reformat numbers to align them, such as in a compact multiline array or table case.  

For instance, start this input:
```
[
  1e500, 1e-500, 4.0
]
```

Now reformat, using the settings, `DontJustifyNumbers=false`, `MaxInlineComplexity=0`.  The result is this:
```
[
    Infinity,        0,        4
]
```

That's not valid JSON, which is a pretty bad mistake for a JSON formatter.

The following input creates similar problems:
```
[
    { "a": 1e500 },
    { "a": 1e-500 },
    { "a": 4.0 }
]
```

It gets reformatted as:
```
[
    {"a": Infinity},
    {"a":        0},
    {"a":        4}
]
```

This problem exists in the C# and JavaScript libraries, as well as the web-base formatter and the Visual Studio Code extension.

**Comment by josdejong** [comment]
Thanks for opening this issue. Maybe good to add dealing with a value having more than 16 digits, like `9123372036854000123`, that will lose the last few digits (at least in JavaScript).

I've had similar issues in my web application and implemented [lossless-json](https://github.com/josdejong/lossless-json) to solve this (and there are quite a few similar `BigNumber` or `BigInt` libraries).

Some ideas for a solution:

1. Implement a function that keeps the numeric values as a string and still is able to format them. I think it is only appending trailing zeros and a dot when needed, if that is indeed the case it sounds like a nice approach to me.
2. Implement a hook to allow you to format/stringify BigNumber values yourself, similar to say the `reviver` and `reducer` arguments of JavaScript's `JSON.parse` and `JSON.stringify` that allow you to deal with custom classes for example.

**Comment by josdejong** [comment]
Ah, thanks for the pointer. So, would you like to alter the exponent like `e35` too, or leave the exponent as is? In the latter case, you can split the value in two parts and format the first part without the exponent.

**Comment by j-brooke** [comment]
Fixed in versions 3.1.1 of the .NET and JS libraries.  Fixed in version 3.1.2 of the VS Code extension.

**Comment by josdejong** [comment]
Thanks!
